metrics/exporter.go

```diff
@@ -64,17 +65,31 @@ type ExporterOptions struct {
 	// See https://github.com/knative/serving/blob/master/config/config-observability.yaml
 	// for details.
 	ConfigMap map[string]string
+
+	// A lister for Secrets to allow dynamic configuration of outgoing TLS client cert.
+	Secrets clientv1.SecretLister `json:"-"`
 }
 
 // UpdateExporterFromConfigMap returns a helper func that can be used to update the exporter
 // when a config map is updated.
-func UpdateExporterFromConfigMap(component string, logger *zap.SugaredLogger) func(configMap *corev1.ConfigMap) {
+// DEPRECATED: Callers should migrate to UpdateExporterFromConfigMap3 (which will be renamed
+// when all callers are migrated).
+func UpdateExporterFromConfigMap(component string, secrets clientv1.SecretLister, logger *zap.SugaredLogger) func(configMap *corev1.ConfigMap) {
+	return UpdateExporterFromConfigMap3(component, secrets, logger)
+}
+
+// UpdateExporterFromConfigMap3 includes a corev1.SecretLister which is used to configure
```

Comment:
Why `3` and not `2` or `Ex` or `Temp`?

Reply:
Because it has 3 arguments.

Reply:
But the old one also had 3 args, no? :-)